libsync/TreeTopology.cpp

```diff
@@ -144,59 +142,71 @@ ssize_t TreeTopology::getParentNodeIndex(ssize_t const& _nodeIndex)
  *  3. _nodeIndex: the index of the node that need select parent from given peers
  */
 void TreeTopology::selectParentNodes(std::shared_ptr<dev::h512s> _selectedNodeList,
-    std::shared_ptr<std::set<dev::h512>> _peers, int64_t const& _nodeIndex)
+    std::shared_ptr<std::set<dev::h512>> _peers, int64_t const& _nodeIndex,
+    int64_t const& _startIndex)
 {
-    ssize_t parentIndex = getParentNodeIndex(_nodeIndex);
+    ssize_t parentIndex = (_nodeIndex - 1) / m_treeWidth;
     // the parentNode is the node-slef
     if (parentIndex == _nodeIndex)
     {
         return;
     }
     dev::h512 selectedNode;
-    while (parentIndex >= m_startIndex)
+    while (parentIndex > 0)
     {
         // find the parentNode from the peers
-        if (getNodeIDByIndex(selectedNode, parentIndex) && _peers->count(selectedNode))
+        auto selectedIndex = getSelectedNodeIndex(parentIndex, _startIndex);
+        if (getNodeIDByIndex(selectedNode, selectedIndex) && _peers->count(selectedNode))
         {
             _selectedNodeList->push_back(selectedNode);
-            TREE_LOG(TRACE) << LOG_DESC("selectParentNodes") << LOG_KV("parentIndex", parentIndex)
-                            << LOG_KV("selectedNode", selectedNode.abridged())
-                            << LOG_KV("idx", m_consIndex);
             break;
         }
-        if (parentIndex == m_startIndex)
+        if (parentIndex <= 0)

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Opposite inner 'if' condition leads to a dead code block.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4727062)